docs/markdown/snippets/add_optimization_plain_option.md

```diff
@@ -0,0 +1,6 @@
+## Add `optimization` `plain` option
```

Comment:
Apparently, this patch causes [`-Oplain` to be passed when building systemd](https://github.com/NixOS/nixpkgs/pull/171069#issuecomment-1186609810). I tried looking where it could be set to that value but could not find any such place.

Reply:
Turns out systemd is passing the value of the meson’s optimization flag directly to cc in custom build target:

https://github.com/systemd/systemd/blob/e228d48b9ebff665e8b1dd1572c53217075c1a5a/src/boot/efi/meson.build#L224

I would probably just mention this specifically in the release notes and move on.